micro-rdk/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 // TODO this check is disabled but shouldn't
 #![allow(clippy::arc_with_non_send_sync)]
+#![cfg_attr(feature = "esp32", feature(linkage))]
```

Comment:
is it a nightly only feature? if so then we have to feature-gate it but let's leave a comment so we know why

Reply:
yes that's why I'll leave a comment